src/conversions.jl

```diff
@@ -70,7 +70,8 @@ end
 # no-op and element-type conversions, plus conversion to and from transparency
 # Colorimetry conversions are in Colors.jl
 convert(::Type{C}, c::C) where {C<:Colorant} = c
-convert(::Type{C}, c) where {C<:Colorant} = cconvert(ccolor(C, typeof(c)), c)
+convert(::Type{C}, c::Colorant) where {C<:Colorant} = cconvert(ccolor(C, typeof(c)), c)
+convert(::Type{C}, c::Number) where {C<:Colorant} = cconvert(ccolor(C, typeof(c)), c)
 cconvert(::Type{C}, c::C) where {C} = c
```

Comment:
BTW, I think this method is dead, based on the current `ccolor` implementation. (`convert(::Type{C}, c::C) where {C<:Colorant}` has priority.) The re-examination of the conversion methods may be necessary.

Reply:
I tried deleting them but without other changes they are still necessary. Any other cleanup can happen in a separate PR.